### PR TITLE
Prelude SIEM: Bump version 5.2.0

### PR DESCRIPTION
**Pull request by ToToL**
Bump version 5.2.0

Add compatibility to Python 3.8

### CODE REVIEW

**Comment by ToToL** [comment]
Hello

For the last QA issue, I don't know how to fix NonsolvableDepsInDev

Can you help me ?

PS : Wrong tag, commits are signed off if I well understand what I have to do

Thanks

Regards

**Comment by vaukai** [comment]
> For the last QA issue, I don't know how to fix NonsolvableDepsInDev
> 
> Can you help me ?
Guess it's because dev-python/croniter has only ( python3_{6,7} ).  Should be ( python3_{6..9} )

**Comment by thesamesam** [comment]
Add "[please reassign]" to the start of the PR name to make it rescan.

For the QA issue, see what @vaukai said. prewikka cannot yet have newer Pythons because croniter didn't, but does now! (It was added about 5 minutes ago).

So, rebase your branch and force push, it should recheck it.

**Comment by ToToL** [comment]
Hello

Thanks for croniter !

I did what you tell me to do. Waiting for rescan.

Regards

File: app-admin/prelude-manager/prelude-manager-5.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	default
+
+	rm -rv "${ED}/run" || die "rm failed"
+	keepdir /var/spool/prelude-manager{,/failover,/scheduler}
+	keepdir /var/spool/prelude/prelude-manager{,/failover,/scheduler}
+
+	find "${D}" -name '*.la' -delete || die
+
+	systemd_dounit "${FILESDIR}/${PN}.service"
+	systemd_newtmpfilesd "${FILESDIR}/${PN}.run" "${PN}.conf"
```
Comment (juippis) [inline app-admin/prelude-manager/prelude-manager-5.2.0.ebuild:66]: This is deprecated and moved to *tmpfiles.eclass*.

File: app-admin/prelude-manager/prelude-manager-5.2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+inherit autotools systemd
+
+DESCRIPTION="Bus communication for all Prelude modules"
+HOMEPAGE="https://www.prelude-siem.org"
+SRC_URI="https://www.prelude-siem.org/pkg/src/5.2.0/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="dbx geoip tcpwrapper xml snmp"
```
Comment (juippis) [inline app-admin/prelude-manager/prelude-manager-5.2.0.ebuild:15]: Could you please sort this list alphabetically while at it.

File: dev-libs/libprelude/libprelude-5.2.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+USE_RUBY="ruby24 ruby25 ruby26 ruby27"
```
Comment (juippis) [inline dev-libs/libprelude/libprelude-5.2.0.ebuild:7]: Ruby-2.4 doesn't exist in the tree anymore.

File: net-analyzer/prelude-lml/prelude-lml-5.2.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	default
+
+	rm -rv "${ED}/run" || die "rm failed"
+	keepdir /var/${PN}
+	keepdir /var/spool/prelude/prelude-lml
+
+	find "${D}" -name '*.la' -delete || die
+
+	systemd_dounit "${FILESDIR}/${PN}.service"
+	systemd_newtmpfilesd "${FILESDIR}/${PN}.run" "${PN}.conf"
```
Comment (juippis) [inline net-analyzer/prelude-lml/prelude-lml-5.2.0.ebuild:57]: Same here about *tmpfiles.eclass*.

**Review by juippis — COMMENTED** [review]
What a massive PR, couple minor issues but looks really good overall.

**Comment by ToToL** [comment]
Hello

Thanks for your review.

All your comments are taken in account.

For spliting, is it really necessary ? Because all packages depends on libprelude, some on libpreludedb, others on prelude-lml, etc. If it's possible, I prefere let it as it is. If not, I will split this.

Thanks

Regards

File: app-admin/prelude-manager/prelude-manager-5.2.0.ebuild (hunk 1 above)
Comment (ToToL) [reply app-admin/prelude-manager/prelude-manager-5.2.0.ebuild]: Done

File: app-admin/prelude-manager/prelude-manager-5.2.0.ebuild (hunk 2 above)
Comment (ToToL) [reply app-admin/prelude-manager/prelude-manager-5.2.0.ebuild]: Done

File: dev-libs/libprelude/libprelude-5.2.0.ebuild (hunk 3 above)
Comment (ToToL) [reply dev-libs/libprelude/libprelude-5.2.0.ebuild]: Done

File: net-analyzer/prelude-lml/prelude-lml-5.2.0.ebuild (hunk 4 above)
Comment (ToToL) [reply net-analyzer/prelude-lml/prelude-lml-5.2.0.ebuild]: Done

File: www-apps/prewikka/prewikka-5.2.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	dev-python/werkzeug[${PYTHON_USEDEP}]
+	media-fonts/font-xfree86-type1"
+
+DEPEND="${RDEPEND}"
+
+BDEPEND="dev-python/lesscpy[${PYTHON_USEDEP}]
+	sys-devel/gettext"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-5.2.0-fix_siteconfig.patch"
```
Comment (juippis) [inline www-apps/prewikka/prewikka-5.2.0.ebuild:41]: Looks like you forgot to push this patch, 
```
 * ERROR: www-apps/prewikka-5.2.0::testworld failed (prepare phase):
 *   patch -p1  failed with /var/tmp/portage/www-apps/prewikka-5.2.0/files/prewikka-5.2.0-fix_siteconfig.patch
 * 
 * Call stack:
 *               ebuild.sh, line  125:  Called src_prepare
 *             environment, line 2951:  Called distutils-r1_src_prepare
 *             environment, line 1214:  Called distutils-r1_python_prepare_all
 *             environment, line 1121:  Called default
 *      phase-functions.sh, line  855:  Called default_src_prepare
 *      phase-functions.sh, line  920:  Called __eapi6_src_prepare
 *             environment, line  218:  Called eapply '/var/tmp/portage/www-apps/prewikka-5.2.0/files/prewikka-5.2.0-fix_siteconfig.patch'
 *             environment, line 1482:  Called _eapply_patch '/var/tmp/portage/www-apps/prewikka-5.2.0/files/prewikka-5.2.0-fix_siteconfig.patch'
 *             environment, line 1420:  Called __helpers_die 'patch -p1  failed with /var/tmp/portage/www-apps/prewikka-5.2.0/files/prewikka-5.2.0-fix_siteconfig.patch'
 *   isolated-functions.sh, line  112:  Called die
 * The specific snippet of code:
 *   		die "$@"
 * 
 * If you need support, post the output of `emerge --info '=www-apps/prewikka-5.2.0::testworld'`,
 * the complete build log and the output of `emerge -pqv '=www-apps/prewikka-5.2.0::testworld'`.
 * The complete build log is located at '/var/tmp/portage/www-apps/prewikka-5.2.0/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/www-apps/prewikka-5.2.0/temp/environment'.
 * Working directory: '/var/tmp/portage/www-apps/prewikka-5.2.0/work/prewikka-5.2.0'
 * S: '/var/tmp/portage/www-apps/prewikka-5.2.0/work/prewikka-5.2.0'

>>> Failed to emerge www-apps/prewikka-5.2.0, Log file:
```

File: www-apps/prewikka/prewikka-5.2.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+pkg_postinst() {
+	optfeature "Asynchronous DNS" dev-python/twisted[${PYTHON_USEDEP}]
+}
```
Comment (juippis) [inline www-apps/prewikka/prewikka-5.2.0.ebuild:46]: Place this below `src_install`.

**Review by juippis — COMMENTED** [review]
Keep it like this. My impression was more like, it is a massive PR without any big faults. So a compliment!

We like to keep commits to a single PR if they are related, like they are here.

File: www-apps/prewikka/prewikka-5.2.0.ebuild (hunk 6 above)
Comment (ToToL) [reply www-apps/prewikka/prewikka-5.2.0.ebuild]: done

File: www-apps/prewikka/prewikka-5.2.0.ebuild (hunk 5 above)
Comment (ToToL) [reply www-apps/prewikka/prewikka-5.2.0.ebuild:41]: done

**Comment by ToToL** [comment]
> Keep it like this. My impression was more like, it is a massive PR without any big faults. So a compliment!

Thanks :)

I just did the change.

Thanks

Regards

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**